### ISSUE TITLE
[CI] DebMetadataTests test05CheckLintian failure

### ISSUE THREAD

**Issue by romseygeek** [elastic/elasticsearch#58730]
**Build scan**: https://gradle-enterprise.elastic.co/s/b2c54hdwwhm2y/console-log?task=:qa:os:destructiveDistroTest.default-deb

**Repro line**: packaging test, so no repro line

**Reproduces locally?**: packaging test, so no reproduction

**Applicable branches**: master

**Failure history**: Failed on master only, for both Ubuntu and Debian packaging tests.  Has failed in the past (see #52554) due to changes in `ml-cpp` but was fixed in February and has not failed since.

**Failure excerpt**:

```
org.elasticsearch.packaging.test.DebMetadataTests > test05CheckLintian FAILED |  
-- | --
  | org.elasticsearch.packaging.util.Shell$ShellException: Command was not successful: [bash -c lintian --fail-on-warnings /var/lib/jenkins/workspace/elastic+elasticsearch+master+multijob+packaging-tests-unix-gcp/os/debian-10-packaging/distribution/packages/deb/build/distributions/elasticsearch-8.0.0-SNAPSHOT-amd64.deb] |  
  | result: exitCode = [1] stdout = [<<Too large to read: 1039698 bytes>>] stderr = [warning: the authors of lintian do not recommend running it with root privileges! |  
  | warning: --fail-on-warnings is deprecated]
```

**Comment by droberts195** [comment]
I don't think it's `ml-cpp` this time:

```
dave@vm-ubuntu-1804-64-1:~$ lintian --fail-on-warnings elasticsearch-8.0.0-SNAPSHOT-amd64.deb
warning: --fail-on-warnings is deprecated
W: elasticsearch: non-standard-file-perm usr/share/elasticsearch/NOTICE.txt 0664 != 0644
W: elasticsearch: package-installs-java-bytecode usr/share/elasticsearch/lib/tools/security-cli/org/apache/lucene/analysis/miscellaneous/DeDuplicatingTokenFilter$DuplicateTaggingFilter.class
W: elasticsearch: package-installs-java-bytecode usr/share/elasticsearch/lib/tools/security-cli/org/apache/lucene/analysis/miscellaneous/DeDuplicatingTokenFilter.class
W: elasticsearch: package-installs-java-bytecode usr/share/elasticsearch/lib/tools/security-cli/org/apache/lucene/analysis/miscellaneous/DisableGraphAttribute.class
W: elasticsearch: package-installs-java-bytecode ... use --no-tag-display-limit to see all (or pipe to a file/program)
W: elasticsearch: non-standard-file-perm usr/share/lintian/overrides/elasticsearch 0664 != 0644
N: 301 tags overridden (177 errors, 93 warnings, 31 info)
```

Presumably with "--no-tag-display-limit to see all (or pipe to a file/program)" it lists every single Java class's `.class` file, explaining why the output is so huge.

Running it like this avoids the Java bytecode warnings:

```
dave@vm-ubuntu-1804-64-1:~$ lintian --fail-on-warnings --suppress-tags package-installs-java-bytecode elasticsearch-8.0.0-SNAPSHOT-amd64.deb
warning: --fail-on-warnings is deprecated
W: elasticsearch: non-standard-file-perm usr/share/elasticsearch/NOTICE.txt 0664 != 0644
W: elasticsearch: non-standard-file-perm usr/share/lintian/overrides/elasticsearch 0664 != 0644
N: 301 tags overridden (177 errors, 93 warnings, 31 info)
```

I guess the two remaining file permissions warnings should be fixed.

**Comment by mark-vieira** [comment]
@rjernst @williamrandolph any thoughts on this? Looks to be isolated to Ubuntu 18.04 an Debian 10.

**Comment by jakelandis** [comment]
two more cases:
7.8 Debian 10 : https://gradle-enterprise.elastic.co/s/ewtdkfa63qkbc
7.8 Ubuntu 18.04: https://gradle-enterprise.elastic.co/s/2youkxk6yruse

**Comment by droberts195** [comment]
When I saw the warnings I assumed that Lintian was looking _inside_ `.jar` files. But it seems that actually the ES build is now accidentally shipping loose `.class` files not inside `.jar` files, in particular those from Lucene. This is a blocker for the 7.8.1 release because Microsoft installer cannot install files with `$` in the name, like `DeDuplicatingTokenFilter$DuplicateTaggingFilter.class`.

**Comment by droberts195** [comment]
> it seems that actually the ES build is now accidentally shipping loose `.class` files not inside `.jar` files

Confirmed:

```
dave@vm-ubuntu-1804-64-1:~$ dpkg-deb -c elasticsearch-8.0.0-SNAPSHOT-amd64.deb | grep '\.class$' | wc -l
6398
dave@vm-ubuntu-1804-64-1:~$ dpkg-deb -c elasticsearch-8.0.0-SNAPSHOT-amd64.deb | grep '\.class$' | head
-rw-r--r-- root/root     13074 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/joda/time/format/StrictISODateTimeFormat$Constants.class
-rw-r--r-- root/root     10724 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/joda/time/format/StrictISODateTimeFormat.class
-rw-r--r-- root/root      9293 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/NodeConnectionsService$ConnectionTarget.class
-rw-r--r-- root/root      2585 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/DiffableUtils$VIntKeySerializer.class
-rw-r--r-- root/root      3217 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/InternalClusterInfoService$1.class
-rw-r--r-- root/root     10323 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/ClusterChangedEvent.class
-rw-r--r-- root/root       940 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/ClusterStateTaskConfig$Basic.class
-rw-r--r-- root/root       530 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/AckedClusterStateTaskListener.class
-rw-r--r-- root/root     19116 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/SnapshotsInProgress$Entry.class
-rw-r--r-- root/root       847 2020-06-30 11:46 ./usr/share/elasticsearch/lib/tools/security-cli/org/elasticsearch/cluster/ClusterInfoService.class
```

So this issue doesn't imply some weird update to Lintian, but a change to the way Elasticsearch is being packaged for distribution.

Presumably this must have been caused by a commit that was made on 29th June.

**Comment by droberts195** [comment]
The problem with class files should be fixed by #59048.  So we are back to just the minor problem of the two text files with `0664` rather than `0644` permissions.
